src/Orleans/Telemetry/TelemetryManager.cs

```diff
@@ -0,0 +1,259 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using Orleans.Runtime.Configuration;
+
+namespace Orleans.Runtime
+{
+    public class TelemetryManager : ITelemetryClient, IDisposable
```

Comment:
naming: I'd like it to be named `TelemetryProducer`. Since it doesn't really manage anything, all the configuration part is in `TelemetryConfiguration`. This is really just the producing part.

Reply:
In this one I disagree, this hooks up the producer with all the consumers, and allows adding more consumers... It is managing.

Reply:
sure